pkg/util/ratelimiter/ratelimiter.go

```diff
@@ -0,0 +1,52 @@
+package ratelimiter
+
+import (
+	"context"
+	"sync"
+
+	"k8s.io/client-go/util/flowcontrol"
+)
+
+type ReloadableRateLimiter struct {
+	lock        *sync.Mutex
+	rateLimiter flowcontrol.RateLimiter
+}
+
+func (r *ReloadableRateLimiter) TryAccept() bool {
+	return r.get().TryAccept()
+}
+
+func (r *ReloadableRateLimiter) Accept() {
+	r.get().Accept()
+}
+
+func (r *ReloadableRateLimiter) Stop() {
+	r.get().Stop()
+}
+
+func (r *ReloadableRateLimiter) QPS() float32 {
+	return r.get().QPS()
+}
+
+func (r *ReloadableRateLimiter) Wait(ctx context.Context) error {
+	return r.get().Wait(ctx)
+}
+
+func (r *ReloadableRateLimiter) get() flowcontrol.RateLimiter {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	return r.rateLimiter
+}
+
+func (r *ReloadableRateLimiter) Set(limiter flowcontrol.RateLimiter) {
+	r.lock.Lock()
+	defer r.lock.Unlock()
+	r.rateLimiter = limiter
+}
```

Comment:
it probably won't make a huge difference, but can we get away with a RWMutex? https://pkg.go.dev/sync#RWMutex

Reply:
I am not sure if a rw mutex is faster since I don't know how it is implemented in go (the gating and reference counting can have more overhead than just looking up the reference).

Reply:
I will check the performance characteristics.

Reply:
This shouldn't be called too much time, shouldn't it? Only when updating or starting the API. Or, Am I missing something? 
If it is not being called too much time, regular Lock and Unlock should be fine here.

Reply:
I could not find a reproducable benefit for any of the following methods: `RWMutext with RLock` vs. `Mutex with Lock` vs. `always Lock with RWMutex`. 

I will leave it as is.

Reply:
It is up to @davidvossel, the current approach seems ok to me since you did not experienced any performance difference.